File: app/src/main/java/com/otraempresa/android/example/RequestCardAsyncTokenAsyncTask.java
package com.otraempresa.android.example;

import android.content.Context;

import com.kushkipagos.android.KushkiException;
import com.kushkipagos.android.Transaction;

public class RequestCardAsyncTokenAsyncTask extends AbstractRequestCadTokenAsyncTask {

    RequestCardAsyncTokenAsyncTask(Context context) {
        super(context);
    }

    @Override
    protected Transaction requestCardAsyncToken(String email) throws KushkiException {
        return kushkiCardAsync.requestCardAsyncToken(1000,
                "https://return.url",email);
    }

}

File: app/src/main/java/com/kushkipagos/kushkidemo/services/RequestCardAsyncTokenAsyncTask.kt
package com.kushkipagos.kushkidemo.services

import android.content.Context
import com.kushkipagos.exceptions.KushkiException
import com.kushkipagos.kushkidemo.contracts.AbstractRequestCadTokenAsyncTask
import com.kushkipagos.models.Transaction

class RequestCardAsyncTokenAsyncTask: AbstractRequestCadTokenAsyncTask {

    constructor(context: Context): super(context)

    @Throws(KushkiException::class)
    override fun requestCardAsyncToken(email: String): Transaction {
        return _kushki.requestCardAsyncToken(1000.0,
            "https://return.url",email)
    }
}